go/src/koding/klientctl/auth/register.go

```diff
@@ -0,0 +1,326 @@
+				Usage: "Username to register",
+			},
+			cli.StringFlag{
+				Name:  "firstName",
+				Usage: "firstName to register",
+			},
+			cli.StringFlag{
+				Name:  "lastName",
+				Usage: "lastName to register",
+			},
+			cli.StringFlag{
+				Name:  "password",
+				Usage: "Username password to register",
+			},
+			cli.StringFlag{
+				Name:  "email",
+				Usage: "Email Address to register",
+			},
+			cli.StringFlag{
+				Name:  "team",
+				Usage: "Team Name to register",
+			},
+			cli.StringFlag{
+				Name:  "company",
+				Usage: "Team Company Name to register (default: team name)",
+			},
+			cli.BoolFlag{
+				Name:  "newsletter",
+				Usage: "Do you want to get occasional newsletters from Koding?",
+			},
+			cli.BoolFlag{
+				Name:  "alreadyMember",
+				Usage: "Do you already registered to Koding?",
+			},
+		},
+	}
+}
+
+const (
+	trueValue  = "true"
+	falseValue = "false"
+)
+
+// RegisterRequest holds the registration request to Koding.
+type RegisterRequest struct {
+	Username        string `json:"username"`
+	FirstName       string `json:"firstName"`
+	LastName        string `json:"lastName"`
+	Password        string `json:"password"`
+	PasswordConfirm string `json:"passwordConfirm"`
+	Email           string `json:"email"`
+	Slug            string `json:"slug"`
+	CompanyName     string `json:"companyName"`
+	Newsletter      string `json:"newsletter,string"`
+	AlreadyMember   string `json:"alreadyMember,string"`
+	Agree           string `json:"agree"`
+}
+
+// RegisterCommand displays version information like Environment or Kite Query ID.
+func RegisterCommand(c *cli.Context, log logging.Logger, _ string) int {
+	rr := initRegisterRequest(c)
+	r, err := checkAndAskRequiredFields(rr)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "Register failed with error:", err)
+		log.Error("%s", err)
+		return 1
+	}
+
+	host := config.Konfig.KodingBaseURL()
+
+	client := httputil.DefaultRestClient(false)
+
+	// TODO ~mehmetali
+	// handle --alreadyMember flag with various option.
+	// There might be some situations that errors need to be ignored
+	token, err := doRegisterRequest(r, client, host)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "Register failed with error:", err)
+		log.Error("%s", err)
+		return 1
+	}
+
+	clientID, err := doLoginRequest(client, host, token)
+	if err != nil {
+		// we don't need to inform user about the error after user registered successfully
+		log.Error("%s", err)
+		return 1
+	}
+
+	// Set clientId into the kd.bolt
+	if err = kloud.Cache().Set("clientId", clientID); err != nil {
+		log.Error("error while caching clientId")
+		return 1
+	}
+
+	// team cannot be empty (because of required while registering)
+	// otherwise it return error while registering user
+	// store groupName or slug as "team" inside the cache
+	if err = kloud.Cache().Set("team", c.String("team")); err != nil {
+		log.Error("error while caching team")
+		return 1
+	}
+
+	fmt.Println("This is your session ID, keep it safe.", clientID)
```

Comment:
We should not print the session ID at all, it should be used internally only. There's no place user can input it. Instead we cache it per team.

See: https://github.com/koding/koding/blob/master/go/src/koding/klientctl/endpoint/auth/auth.go#L70